l2gv2/anomaly_detection.py

```diff
@@ -1,88 +1,115 @@
-from tqdm.notebook import tqdm
-import scipy.sparse as ss
-import scipy.sparse.linalg as sl
-import community
-
-import torch_geometric as tg
-import torch_scatter as ts
-from torch_geometric.nn import GCNConv
-from torch_geometric.data import Data
-from torch_geometric.utils.convert import from_networkx
-from torch_geometric.utils import to_networkx
-from torch_geometric.transforms import LargestConnectedComponents
-
-import local2global as l2g
-import local2global_embedding
-from local2global_embedding import patches, clustering
-from local2global_embedding.network import graph
-from local2global_embedding.network import TGraph
-from local2global import Patch
-import local2global_embedding.embedding.svd as svd
-import local2global_embedding.embedding.gae as gae
-import local2global_embedding.patches as patches
-
-import numpy as np 
-import pandas as pd 
-import torch 
-import torch.nn.functional as F
-import torch.optim as optim
-import optuna
-from optuna.trial import TrialState
-
-from sklearn.metrics import roc_auc_score, f1_score, accuracy_score, precision_score, recall_score, confusion_matrix
-from sklearn.model_selection import train_test_split
-from sklearn.model_selection import StratifiedKFold
-#from google.colab import drive, files
-import networkx as nx
-import matplotlib.pyplot as plt
+"""Anomaly detection module."""
+
+import numpy as np
 
 
 def raw_anomaly_score_node_patch(aligned_patch_emb, emb, node):
```

Comment:
Optional for this PR - ideally parameters would be typed, and docs can take parameter types from here. Also helps with type checking with e.g. mypy or LSP hinting